packages/jupyter-ai/jupyter_ai/chat_handlers/base.py

```diff
@@ -89,7 +112,13 @@ def __init__(
         self._root_chat_handlers = root_chat_handlers
         self.model_parameters = model_parameters
         self._chat_history = chat_history
-        self.parser = argparse.ArgumentParser()
+        self.parser = argparse.ArgumentParser(
+            add_help=False, description=self.help, formatter_class=MarkdownHelpFormatter
+        )
+        # the default help would exit; instead implement a custom help
+        self.parser.add_argument(
+            "-h", "--help", action="store_true", help="show this help message"
+        )
```

Comment:
Also, maybe we should only add `-h/--help` for every slash command except `/help`? It doesn't really make sense to support `/help --help`.

Try this:
```suggestion
        if not self.__class__.id == "help":
            self.parser.add_argument(
                "-h", "--help", action="store_true", help="show this help message"
            )
```